app/lib/dartdoc/dartdoc_runner.dart

```diff
@@ -19,7 +28,68 @@ class DartdocRunner implements TaskRunner {
 
   @override
   Future<bool> runTask(Task task) async {
-    // TODO: implement doc generation and upload
-    return false; // no race detected
+    final tempDir =
+        await Directory.systemTemp.createTemp('pub-dartlang-dartdoc');
+    final pubCacheDir = await tempDir.resolveSymbolicLinks();
+
+    final pubEnv = new PubEnvironment(pubCacheDir: pubCacheDir);
+
+    try {
+      final pkgLocation =
+          await pubEnv.getLocation(task.package, version: task.version);
```

Comment:
Does this end up downloading from  the proper `https://pub.dartlang.org/packages/<name>/versions/<version>.tar.gz` url?

(so we don't have the memcache issue)

Reply:
Unfortunately `pub` does a sanity check before doing that:
https://github.com/dart-lang/pub/blob/master/lib/src/command/cache_add.dart#L57
https://github.com/dart-lang/pub/blob/master/lib/src/source/hosted.dart#L130

Btw. this affects `pana` the very same way. /cc @kevmoo